### DISCUSSION TITLE
getTag API or normalize tags

### DISCUSSION THREAD

**Discussion by MichaelLeeHobbs** [mliebelt/pgn-parser#85] [Ideas]
Due to variations in casing it would be useful to have `getTag` function.

```javascript
// reference code
const getTag = (game, tag) => {
  if (game.tags[tag]) return game.tags[tag]
  const keys = Object.keys(game.tags)
  const tagRegex = new RegExp(tag, 'i')
  const key = keys.find((key)=>tagRegex.test(key))
  return game.tags[key]
}
```

```javascript
// code to parse
// ...

const WhiteElo = games[0].tags.getTag('whiteElo')  // getTag already has a refence to tags so game should not be needed to be passed

```

Another option would be to normalize know/standard tags.

If you want I could setup a pull request for this ie getTag with test. Normalizing I'd have to take a look first.

**Comment by mliebelt** [comment]
The grammar should return the tags normalized, and it allows only some variations (being liberal with first upper-/lowercase). So you should already get the right tag from the grammar.

**Reply by MichaelLeeHobbs** [reply]
I've seen WhiteElo and WhiteELO which is what lead to this.

**Reply by mliebelt** [reply]
From the grammar:

`'WhiteELO' / 'WhiteElo' / 'Whiteelo' / 'whiteelo' / 'whiteElo'` are all supported out of the box.

And the rules are (which is wrong):

```
	/ whiteEloKey ws value:integerOrDashString  { return { name: 'WhiteELO', value: value }; }
	/ blackEloKey ws value:integerOrDashString  { return { name: 'BlackELO', value: value }; }
```

The [specification](https://github.com/mliebelt/pgn-spec-commented/blob/main/pgn-specification.md#912-tags-whiteelo-blackelo) tells that is should be `WhiteElo` and `BlackElo`. I will file a bug for this, and fix it then.

**Reply by mliebelt** [reply]
See #86 (which was done long ago), I will therefore close that discussion.
